pyretailscience/analysis/composite_rank.py

```diff
@@ -153,46 +162,115 @@ def __init__(
             ...     agg_func="mean"
             ... )
             >>> # Products with lowest composite_rank should be reviewed for delisting
+
+            >>> # Group-based ranking: Rank products within each category
+            >>> ranker = CompositeRank(
+            ...     df=product_data,
+            ...     rank_cols=[
+            ...         ("weekly_sales", "desc"),
+            ...         ("margin_percentage", "desc"),
+            ...         ("stock_cover_days", "asc")
+            ...     ],
+            ...     agg_func="mean",
+            ...     group_col="product_category"
+            ... )
+            >>> # Electronics products ranked against other electronics
+            >>> # Apparel products ranked against other apparel
         """
         if isinstance(df, pd.DataFrame):
             df = ibis.memtable(df)
 
-        # Validate columns and sort orders
-        valid_sort_orders = ["asc", "ascending", "desc", "descending"]
-
-        rank_mutates = {}
-        for col_spec in rank_cols:
-            if isinstance(col_spec, str):
-                col_name = col_spec
-                sort_order = "asc"
-            else:
-                if len(col_spec) != 2:  # noqa: PLR2004 - Error message below explains the value
-                    msg = (
-                        f"Column specification must be a string or a tuple of (column_name, sort_order). Got {col_spec}"
-                    )
+        self._validate_group_col(group_col, df)
+        rank_mutates = self._process_rank_columns(rank_cols, df, group_col, ignore_ties)
+        df = df.mutate(**rank_mutates)
+        self.table = self._create_composite_ranking(df, rank_mutates, agg_func)
+
+    def _validate_group_col(self, group_col: str | list[str] | list | None, df: ibis.Table) -> None:
```

Comment:
List shouldn't be there. It should always be a list of strings
```suggestion
    def _validate_group_col(self, group_col: str | list[str] | None, df: ibis.Table) -> None:
```